airflow/serialization/serialized_objects.py

```diff
@@ -444,14 +446,14 @@ class DependencyDetector:
     @staticmethod
     def detect_task_dependencies(task: BaseOperator) -> Optional['DagDependency']:
         """Detects dependencies caused by tasks"""
-        if task.task_type == "TriggerDagRunOperator":
+        if isinstance(task, TriggerDagRunOperator):
```

Comment:
Since this is done during serialization time this should be ok, however can you please add test(s)

Reply:
Sure I'll look into how tests work in airflow, and add one, I suppose, for testing that both `TriggerDagRunOperator` and inherited classes give the expected result.

Reply:
@kaxil do you think that perhaps there should be an attribute on task sensor (like a class attribute that is set on it, and which could also be set in other implementations) that we look at (instead of the class itself) so that there could be other implementations of task-sensor-like operators which do not inherit from external task sensor but also have dag dependencies?

so that it needn't necessarily be a subclass.  maybe it's a bit too magical, and just thinking aloud, but perhaps we could look at the arguments of the operator to infer whether DagDependency should be created?